spec/requests/providers/link_case/searches_controller_spec.rb

```diff
@@ -0,0 +1,116 @@
+      let(:linked_application) { create(:linked_application, lead_application_id: linkable_application.id, associated_application_id: legal_aid_application.id) }
+
+      it "clears application on page load" do
+        linked_application
+        expect { get_request }.to change(LinkedApplication, :count).from(1).to(0)
+      end
+    end
+  end
+
+  describe "PATCH /providers/applications/:legal_aid_application_id/link_case/search" do
+    subject(:patch_request) { patch providers_legal_aid_application_link_case_search_path(legal_aid_application), params: }
+
+    context "when valid search term is entered" do
+      let(:params) { { linked_application: { search_ref: "L-TVH-U0T" } } }
+
+      before { linkable_application }
+
+      it "redirects to the linking case confirmation page" do
+        patch_request
+        expect(response).to redirect_to(providers_legal_aid_application_link_case_confirmation_path(legal_aid_application))
+      end
+
+      it "creates a linked application" do
+        expect { patch_request }.to change(LinkedApplication, :count).from(0).to(1)
+      end
+    end
+
+    context "when search term is blank" do
+      let(:params) { { linked_application: { search_ref: "" }, continue_button: "Save and continue" } }
+
+      it "stays on the page and shows a validation error" do
+        patch_request
+        expect(response).to have_http_status(:ok)
+        expect(page).to have_error_message("Enter an application reference to search for")
+      end
+
+      it "does not create a linked application" do
+        expect { patch_request }.not_to change(LinkedApplication, :count).from(0)
+      end
+    end
+
+    context "when invalid search term is entered" do
+      let(:params) { { linked_application: { search_ref: "testing" }, continue_button: "Save and continue" } }
+
+      it "stays on the page and shows a validation error" do
+        patch_request
+        expect(response).to have_http_status(:ok)
+        expect(page).to have_error_message("Enter a valid application reference to search for")
+      end
+
+      it "does not create a linked application" do
+        expect { patch_request }.not_to change(LinkedApplication, :count).from(0)
+      end
+    end
+
+    context "when application reference was not found" do
+      let(:params) { { linked_application: { search_ref: "L-17B-RT1" }, continue_button: "Save and continue" } }
+
+      it "stays on the page and shows a validation error" do
+        patch_request
+        expect(response).to have_http_status(:ok)
+        expect(page).to have_error_message("The application reference entered cannot be found")
+      end
+
+      it "does not create a linked application" do
+        expect { patch_request }.not_to change(LinkedApplication, :count).from(0)
+      end
+    end
+
+    context "when form submitted using Save as draft button" do
+      let(:params) { { linked_application: { search_ref: "L-TVH-U0T" }, draft_button: "Save and come back later" } }
```

Comment:
the value supplied to `draft_button` is irrelevant, so in the interests of having tests reflect that I tend to add something that makes that clear(er), either ""/blank or "somewordtoclarifyirrelevance"
```suggestion
      let(:params) { { linked_application: { search_ref: "L-TVH-U0T" }, draft_button: "irrelevant" } }
```